### PR TITLE
Pass selected dates into url generator.

### PR DESCRIPTION
**Pull request by danielgent**
## Summary

Addresses issue #2983

## Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: assets/js/modules/analytics/components/module/ModuleAcquisitionChannelsWidget/Header.js [hunk 1]
```diff
@@ -26,14 +26,20 @@ import { Fragment } from '@wordpress/element';
...
+import { MODULES_ANALYTICS, DATE_RANGE_OFFSET } from '../../../datastore/constants';
 import WidgetHeaderTitle from '../../../../../googlesitekit/widgets/components/WidgetHeaderTitle';
 import WidgetHeaderCTA from '../../../../../googlesitekit/widgets/components/WidgetHeaderCTA';
 import { CORE_USER } from '../../../../../googlesitekit/datastore/user/constants';
+import { generateDateRangeArgs } from '../../../util/report-date-range-args';
 const { useSelect } = Data;
 
 export default function Header() {
-	const url = useSelect( ( select ) => select( MODULES_ANALYTICS ).getServiceReportURL( 'trafficsources-overview' ) );
+	const { startDate, endDate } = useSelect( ( select ) => select( CORE_USER ).getDateRangeDates( {
```
Comment (eugene-manuilov) [inline assets/js/modules/analytics/components/module/ModuleAcquisitionChannelsWidget/Header.js:37]: Destructing is redundant here, we can simply use the classical approach:

```javascript
const dates = useSelect( ( select ) => select( CORE_USER ).getDateRangeDates( { ... } );
const url = useSelect( ( select ) => select( MODULES_ANALYTICS ).getServiceReportURL( 'trafficsources-overview', generateDateRangeArgs( dates ) ) );
```

File: assets/js/modules/analytics/components/module/ModuleAcquisitionChannelsWidget/Header.js [hunk 2]
```diff
@@ -26,14 +26,20 @@ import { Fragment } from '@wordpress/element';
...
+		offsetDays: DATE_RANGE_OFFSET,
+	} ) );
+	const url = useSelect( ( select ) => select( MODULES_ANALYTICS ).getServiceReportURL( 'trafficsources-overview',
+		generateDateRangeArgs( { startDate, endDate } ) )
+	);
```
Comment (eugene-manuilov) [inline assets/js/modules/analytics/components/module/ModuleAcquisitionChannelsWidget/Header.js:42]: Either all arguments should be on separate lines or none:

```javascript
const url = useSelect( ( select ) => select( MODULES_ANALYTICS ).getServiceReportURL(
    'trafficsources-overview',
    generateDateRangeArgs( { startDate, endDate } ),
) );
```
```javascript
const url = useSelect( ( select ) => select( MODULES_ANALYTICS ).getServiceReportURL( 'trafficsources-overview', generateDateRangeArgs( { startDate, endDate } ) ) );
```

File: assets/js/modules/analytics/components/module/ModuleOverviewWidget/Header.js [hunk 3]
```diff
@@ -27,15 +27,20 @@ import { Fragment } from '@wordpress/element';
...
 const { useSelect } = Data;
 
 export default function Header() {
-	const visitorsOverview = useSelect( ( select ) => select( MODULES_ANALYTICS ).getServiceReportURL( 'visitors-overview' ) );
+	const { startDate, endDate } = useSelect( ( select ) => select( CORE_USER ).getDateRangeDates( {
+		offsetDays: DATE_RANGE_OFFSET,
+	} ) );
+	const visitorsOverview = useSelect( ( select ) => select( MODULES_ANALYTICS ).getServiceReportURL( 'visitors-overview',
+		generateDateRangeArgs( { startDate, endDate } ) )
+	);
```
Comment (eugene-manuilov) [inline assets/js/modules/analytics/components/module/ModuleOverviewWidget/Header.js:42]: Same here: destruction is redundant and either all arguments should be separate lines or none.

File: assets/js/modules/analytics/components/module/ModulePopularPagesWidget/Header.js [hunk 4]
```diff
@@ -27,15 +27,21 @@ import { Fragment } from '@wordpress/element';
...
 const { useSelect } = Data;
 
 export default function Header() {
-	const contentPagesURL = useSelect( ( select ) => select( MODULES_ANALYTICS ).getServiceReportURL( 'content-pages' ) );
-	const currentDayCount = useSelect( ( select ) => select( CORE_USER ).getDateRangeNumberOfDays() );
+	const { startDate, endDate } = useSelect( ( select ) => select( CORE_USER ).getDateRangeDates( {
+		offsetDays: DATE_RANGE_OFFSET,
+	} ) );
+	const contentPagesURL = useSelect( ( select ) => select( MODULES_ANALYTICS ).getServiceReportURL( 'content-pages',
+		generateDateRangeArgs( { startDate, endDate } ) )
```
Comment (eugene-manuilov) [inline assets/js/modules/analytics/components/module/ModulePopularPagesWidget/Header.js:41]: Same here.

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]
Thanks, @danielgent! It looks good from the technical point of view, but there are a few formatting issues that we need to address before it can be merged. Could you please take a look at it and fix it?

File: assets/js/modules/analytics/components/module/ModuleAcquisitionChannelsWidget/Header.js (hunk 1 above)
Comment (danielgent) [reply assets/js/modules/analytics/components/module/ModuleAcquisitionChannelsWidget/Header.js:37]: Cool @eugene-manuilov .  There's copying and pasting here so I'll make the same change everywhere

File: assets/js/modules/analytics/components/module/ModuleAcquisitionChannelsWidget/Header.js (hunk 2 above)
Comment (danielgent) [reply assets/js/modules/analytics/components/module/ModuleAcquisitionChannelsWidget/Header.js:42]: Good point @eugene-manuilov !  I copied and pasted the code from elsewhere hoping to maintain consistency.  I think recently I've got used to prettier and eslint sorting out all the formatting and me not thinking :see_no_evil: 

I'll take more care in the future! :+1:

File: assets/js/modules/analytics/components/module/ModuleAcquisitionChannelsWidget/Header.js [hunk 5]
```diff
@@ -26,14 +26,21 @@ import { Fragment } from '@wordpress/element';
...
 
 export default function Header() {
-	const url = useSelect( ( select ) => select( MODULES_ANALYTICS ).getServiceReportURL( 'trafficsources-overview' ) );
+	const dates = useSelect( ( select ) => select( CORE_USER ).getDateRangeDates( {
+		offsetDays: DATE_RANGE_OFFSET,
+	} ) );
+	const url = useSelect( ( select ) => select( MODULES_ANALYTICS ).getServiceReportURL(
+		'trafficsources-overview',
+		generateDateRangeArgs( dates ) )
+	);
```
Comment (eugene-manuilov) [inline assets/js/modules/analytics/components/module/ModuleAcquisitionChannelsWidget/Header.js:43]:
```suggestion
	const url = useSelect( ( select ) => select( MODULES_ANALYTICS ).getServiceReportURL(
		'trafficsources-overview',
		generateDateRangeArgs( dates ),
	) );
```

File: assets/js/modules/analytics/components/module/ModuleOverviewWidget/Header.js [hunk 6]
```diff
@@ -27,15 +27,21 @@ import { Fragment } from '@wordpress/element';
...
 
 export default function Header() {
-	const visitorsOverview = useSelect( ( select ) => select( MODULES_ANALYTICS ).getServiceReportURL( 'visitors-overview' ) );
+	const dates = useSelect( ( select ) => select( CORE_USER ).getDateRangeDates( {
+		offsetDays: DATE_RANGE_OFFSET,
+	} ) );
+	const visitorsOverview = useSelect( ( select ) => select( MODULES_ANALYTICS ).getServiceReportURL(
+		'visitors-overview',
+		generateDateRangeArgs( dates ) )
+	);
```
Comment (eugene-manuilov) [inline assets/js/modules/analytics/components/module/ModuleOverviewWidget/Header.js:43]:
```suggestion
	const visitorsOverview = useSelect( ( select ) => select( MODULES_ANALYTICS ).getServiceReportURL(
		'visitors-overview',
		generateDateRangeArgs( dates ),
	) );
```

File: assets/js/modules/analytics/components/module/ModulePopularPagesWidget/Header.js [hunk 7]
```diff
@@ -27,15 +27,22 @@ import { Fragment } from '@wordpress/element';
...
-	const contentPagesURL = useSelect( ( select ) => select( MODULES_ANALYTICS ).getServiceReportURL( 'content-pages' ) );
-	const currentDayCount = useSelect( ( select ) => select( CORE_USER ).getDateRangeNumberOfDays() );
+	const dates = useSelect( ( select ) => select( CORE_USER ).getDateRangeDates( {
+		offsetDays: DATE_RANGE_OFFSET,
+	} ) );
+	const contentPagesURL = useSelect( ( select ) => select( MODULES_ANALYTICS ).getServiceReportURL(
+		'content-pages',
+		generateDateRangeArgs( dates ) )
 
+	);
```
Comment (eugene-manuilov) [inline assets/js/modules/analytics/components/module/ModulePopularPagesWidget/Header.js:44]:
```suggestion
	const contentPagesURL = useSelect( ( select ) => select( MODULES_ANALYTICS ).getServiceReportURL(
		'content-pages',
		generateDateRangeArgs( dates ),
	) );
```

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]
A few more minor issues and it should be good.

File: assets/js/modules/analytics/components/module/ModuleAcquisitionChannelsWidget/Header.js [hunk 8]
```diff
@@ -26,14 +26,21 @@ import { Fragment } from '@wordpress/element';
...
 
 export default function Header() {
-	const url = useSelect( ( select ) => select( MODULES_ANALYTICS ).getServiceReportURL( 'trafficsources-overview' ) );
+	const dates = useSelect( ( select ) => select( CORE_USER ).getDateRangeDates( {
+		offsetDays: DATE_RANGE_OFFSET,
+	} ) );
+	const url = useSelect( ( select ) => select( MODULES_ANALYTICS ).getServiceReportURL(
+		'trafficsources-overview',
+		generateDateRangeArgs( dates )
+	)	);
```
Comment (eugene-manuilov) [inline assets/js/modules/analytics/components/module/ModuleAcquisitionChannelsWidget/Header.js:43]:
```suggestion
	) );
```

File: assets/js/modules/analytics/components/module/ModuleOverviewWidget/Header.js [hunk 9]
```diff
@@ -27,15 +27,21 @@ import { Fragment } from '@wordpress/element';
...
 
 export default function Header() {
-	const visitorsOverview = useSelect( ( select ) => select( MODULES_ANALYTICS ).getServiceReportURL( 'visitors-overview' ) );
+	const dates = useSelect( ( select ) => select( CORE_USER ).getDateRangeDates( {
+		offsetDays: DATE_RANGE_OFFSET,
+	} ) );
+	const visitorsOverview = useSelect( ( select ) => select( MODULES_ANALYTICS ).getServiceReportURL(
+		'visitors-overview',
+		generateDateRangeArgs( dates )
+	)	);
```
Comment (eugene-manuilov) [inline assets/js/modules/analytics/components/module/ModuleOverviewWidget/Header.js:43]:
```suggestion
	) );
```

File: assets/js/modules/analytics/components/module/ModulePopularPagesWidget/Header.js [hunk 10]
```diff
@@ -27,15 +27,21 @@ import { Fragment } from '@wordpress/element';
...
 
 export default function Header() {
-	const contentPagesURL = useSelect( ( select ) => select( MODULES_ANALYTICS ).getServiceReportURL( 'content-pages' ) );
+	const dates = useSelect( ( select ) => select( CORE_USER ).getDateRangeDates( {
+		offsetDays: DATE_RANGE_OFFSET,
+	} ) );
+	const contentPagesURL = useSelect( ( select ) => select( MODULES_ANALYTICS ).getServiceReportURL(
+		'content-pages',
+		generateDateRangeArgs( dates )
+	)	);
```
Comment (eugene-manuilov) [inline assets/js/modules/analytics/components/module/ModulePopularPagesWidget/Header.js:43]:
```suggestion
	) );
```

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]
Ok, the last three changes. I'll apply it.

**Review by eugene-manuilov — APPROVED** [review]

**Pull request merged**